Counterexamples/VitaliSetNotMeasurable.lean

```diff
@@ -0,0 +1,318 @@
+  constructor
+  · rw [← h1]
+    exact volume_mono vitaliUnion_lower_bound
+  · rw [← h2]
+    exact volume_mono vitaliUnion_upper_bound
+
+/-- The shifted copies of vitaliSet in vitaliUnion are pairwise disjoint. -/
+lemma vitali_pairwise_disjoint : vI.PairwiseDisjoint vitaliSet' := by
+  intro x x_vI y y_vI x_ne_y
+  refine Set.disjoint_iff.mpr ?_
+  intro z
+  simp only [mem_inter_iff, mem_empty_iff_false, imp_false, not_and]
+  intro z_x z_y
+  simp only [vitaliSet', vitaliSet, image_add_right, preimage_setOf_eq, mem_setOf_eq] at z_x
+  simp only [vitaliSet', vitaliSet, image_add_right, preimage_setOf_eq, mem_setOf_eq] at z_y
+  have ⟨t_x, rep_x⟩ := z_x
+  have ⟨t_y, rep_y⟩ := z_y
+  have vRep_eq : vS.r (vRep t_x) (vRep t_y) := by
+    simp only [vS, mem_range]
+    simp only [rep_x, rep_y, add_sub_add_left_eq_sub, sub_neg_eq_add]
+    have : x ∈ range ((↑) : ℚ → ℝ) := by exact mem_of_mem_inter_right x_vI
+    have ⟨q_x, q_x_eq⟩ := mem_range.mp (mem_of_mem_inter_right x_vI)
+    have ⟨q_y, q_y_eq⟩ := mem_range.mp (mem_of_mem_inter_right y_vI)
+    use (-q_x + q_y)
+    simp [← q_x_eq, ← q_y_eq]
+  have vT_eq := Quotient.sound vRep_eq
+  simp only [vRep_spec] at vT_eq
+  have x_eq_y : x = y := by
+    calc x = z - ↑(vRep t_x) := by { simp [rep_x] }
+         _ = z - ↑(vRep t_y) := by { simp [vT_eq] }
+         _ = y := by { simp [rep_y] }
+  contradiction
+
+/-- vI is a countable set. -/
+lemma vI_countable : vI.Countable := by
+  refine Countable.mono inter_subset_right ?_
+  apply countable_range
+
+/-- This is the main result that will lead to a contradiction:
+if the vitaliSet is null-measurable, then the volume of vitaliUnion is
+the sum of countably many copies of vitaliSet.  -/
+lemma vitaliUnion_volume_sum (hm : NullMeasurableSet vitaliSet volume) :
+    volume vitaliUnion = ∑' (_ : ↑vI), volume vitaliSet := by
+  have hm' : ∀ i ∈ vI, NullMeasurableSet (vitaliSet' i) volume := by
+    intro i i_vI
+    rw [vitaliSet']
+    apply shift_nullmeasurable hm
+  rw [vitaliUnion, biUnion_volume vI_countable vitali_pairwise_disjoint hm']
+  refine tsum_congr ?_
+  intro i
+  rw [vitaliSet', shift_volume]
+
+/-- vI is an infinite set. -/
+lemma vI_infinite : vI.Infinite := by
+  let f : ℕ → ℝ := fun n ↦ 1 / (n + 1)
+  have f_inj : f.Injective := by { intro m n ; simp [f] }
+  have f_rng_inf : (range f).Infinite := infinite_range_of_injective f_inj
+  refine Set.Infinite.mono ?_ f_rng_inf
+  apply range_subset_iff.mpr
+  intro n
+  simp [f, vI]
+  constructor
+  · simp only [inv_eq_one_div]
+    have h1 : (0 : ℝ) < ↑n + 1 := by
+      have : (0 : ℝ) ≤ ↑n := Nat.cast_nonneg n
+      linarith
+    constructor
+    · rw [le_div_iff₀' h1]
+      linarith
+    · rw [div_le_iff₀' h1]
+      linarith
+  · use ((↑ n + 1)⁻¹)
+    simp
+
+/-- The following theorems are the main results.
+
+vitaliSet is not null-measurable. -/
+theorem vitaliSet_not_nullmeasurable : ¬ (NullMeasurableSet vitaliSet volume) := by
```

Comment:
I think you can prove (and add to Mathlib) an auxiliary theorem for this: a sum `Σ i : I, x` for an infinite type `I` is either `0` or `⊤`.